src/util.ts

```diff
@@ -51,13 +51,39 @@ export function meanArray(y: number[][]) {
 }
 
 export function findClosest(x: number, arr: number[]) {
-    // Using binary search here would be nicer; we'll always have 'x'
-    // sorted, and will do this fairly often.
-    const i = arr.findIndex((el) => el >= x);
+    const i = search(arr, (el) => el >= x);
     if (arr[i] === x || i === 0) {
         return i;
     }
-    const xi = arr[i - 1];
-    const xj = arr[i];
-    return (x - xi) < (xj - x) ? i - 1 : i;
+    if (i === arr.length) {
+        return arr.length - 1;
+    }
+    return (x - arr[i - 1]) < (arr[i] - x) ? i - 1 : i;
+}
+
+export function search(x: number[], pred: (el: number) => boolean) {
+    // Aim is to find the first element that is `true` for this condition.
```

Comment:
Maybe add to comment something to effect that assumption is condition will be true for all subsequent elements too. ?